### PR TITLE
feat: read, validate and build component styles

### PR DESCRIPTION
**Pull request by M-Ryan92**

### CODE REVIEW

**Review by BNNBetty — APPROVED** [review]

File: src/types.ts [hunk 1]
```diff
@@ -44,6 +44,91 @@ export type DefaultCategory =
...
+  [key in StyleStateKeys]?: string[];
+} & { basis: string[] };
+
+export interface StyleDefinitionContentOverwrites
+  extends Omit<StyleDefinitionState, 'name'> {
+  name: string;
+}
+
+export type BuildStyleDefinitionContentOverwrites =
+  Partial<StyleDefinitionContentBase>;
```
Comment (NiekBetty) [inline src/types.ts:115]: Can you be more specific here? 😁

File: src/bb-components-build.ts [hunk 2]
```diff
@@ -28,8 +30,10 @@ import {
   checkOptionCategoryReferences,
 } from './utils/validation';
 import validateComponents from './validations/component';
+import validateStyles from './validations/styles';
 import validateInteractions from './validations/interaction';
 import validatePrefabs from './validations/prefab';
+import { access } from 'fs';
```
[the remark this answers is not in the document]
Comment (M-Ryan92) [reply src/bb-components-build.ts]: check think i deleted this already as the linter failed on this :P

File: __tests__/components-build/styles.test.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  t.deepEqual(buildReferenceStyle(style), style);
+});
+
+test('it build a styleReference for the componentPrefab for states', (t: Context): void => {
+  const overwrite: StyleDefinitionContentOverwrites[] = [
+    {
+      name: 'basis',
+      cssObject: {
```
[the remark this answers is not in the document]
Comment (M-Ryan92) [reply __tests__/components-build/styles.test.tsx:28]: changed to cssObject to content

File: src/types.ts (hunk 1 above)
Comment (M-Ryan92) [reply src/types.ts:115]: the StyleDefinitionContentBase is specified to have all keys present in the object so this is as specific as we could be we want all the keys to be optional as the base requires them all to be present

File: src/types.ts [hunk 4]
```diff
@@ -56,6 +141,57 @@ export interface Component {
...
+      letterSpacing?: string;
+      lineHeight?: string;
+      padding?: string | string[];
+      textDecoration?: string;
+      textTransform?: string;
+    }
+  | BuildStyleDefinitionContentOverwrites;
+
+export type BuildStyle = {
+  name?: string; // TODO: make this required
```
[the remark this answers is not in the document]
Comment (M-Ryan92) [reply src/types.ts:174]: yes will be a fix e should apply after we ship it

**Review by BNNBetty — APPROVED** [review]

**Pull request merged**